### PR TITLE
EQL: Fix custom scripting for functions

### PR DESCRIPTION
**Pull request by costin**
Improve separation of scripting between EQL and SQL by delegating common
methods to QL. The context detection is determined based on the package
to avoid having repetitive class hierarchies.

Relates #53688

### CODE REVIEW

**Review by astefan — APPROVED** [review]
LGTM

**Review by matriv — APPROVED** [review]
LGTM!

**Review by bpintea — APPROVED** [review]
LGTM

**Review by aleksmaus — APPROVED** [review]

File: x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml [hunk 1]
```diff
@@ -881,34 +881,6 @@ file where opcode=0 and indexOf(file_name, 'explorer.', 0) == 0'''
 expected_event_ids  = [88, 92]
 description = "check substring ranges"
 
-[[queries]]
```
Comment (rw-access) [inline x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml]: these should be moved to `test_queries_supported.toml` if they are working now

Comment (aleksmaus) [reply x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml]: they should be already there, test_queries_unsupported.toml file more like a blacklist for test harness

**Pull request merged**